engine/schema/src/main/java/com/cloud/service/dao/ServiceOfferingDaoImpl.java

```diff
@@ -289,4 +294,33 @@ public ServiceOfferingVO findDefaultSystemOffering(String offeringName, Boolean
         }
         return serviceOffering;
     }
+
+    @Override
+    public long removeUniqueName(String uniqueName){
+      SearchCriteria<ServiceOfferingVO> sc = UniqueNameSearch.create();
+      sc.setParameters("name", uniqueName);
+      List<ServiceOfferingVO> vos = search(sc, null, null, false);
+      if (vos.size() != 0) {
+        ServiceOfferingVO seVo = vos.get(0);
+        seVo.setUniqueName(null);
+        seVo.setDefaultUse(false);
+        update(seVo.getId(), seVo);
+        return seVo.getId();
+      } else {
+        return 0;
+      }
+    }
+
+    @Override
+    public void resetUniqueName(long id, String uniqueName){
+      SearchCriteria<ServiceOfferingVO> sc = IdSearch.create();
+      sc.setParameters("id", id);
+      List<ServiceOfferingVO> vos = search(sc, null, null, false);
+      if (vos.size() != 0) {
+        ServiceOfferingVO seVo = vos.get(0);
+        seVo.setUniqueName(uniqueName);
+        seVo.setDefaultUse(true);
```

Comment:
does a reset of uniquename always mean this will be the default offering?

Reply:
I implemented this function so the uniquename and default status of the previously default service offering gets reset when persisting the new one fails.